src/utils/__tests__/resolve_url.test.ts

```diff
@@ -20,16 +20,51 @@ import resolveURL, {
 } from "../resolve_url";
 
 describe("utils - resolveURL", () => {
+  it("should return an empty string if no argument is given", () => {
+    expect(resolveURL()).to.equal("");
+  });
+
   it("should concatenate multiple URLs", () => {
     expect(resolveURL("http://toto.com/a"))
       .to.equal("http://toto.com/a");
     expect(resolveURL("http://toto.com/a", "b/c/d/", "g.a"))
       .to.equal("http://toto.com/a/b/c/d/g.a");
   });
+
+  it("should ignore empty strings when concatenating multiple URLs", () => {
+    expect(resolveURL("", "http://toto.com/a", ""))
+      .to.equal("http://toto.com/a");
+    expect(resolveURL("http://toto.com/a", "b/c/d/", "", "g.a"))
+      .to.equal("http://toto.com/a/b/c/d/g.a");
+  });
+
+  it("should remove a leading slash if one", () => {
+    expect(resolveURL("http://toto.com/a", "/b/c/d/", "/", "/g.a"))
+      .to.equal("http://toto.com/a/b/c/d/g.a");
+  });
+
   it("should reset the concatenation if a given string contains a scheme", () => {
     expect(resolveURL("http://toto.com/a", "b/c/d/", "torrent://g.a", "b"))
       .to.equal("torrent://g.a/b");
   });
+
+  it("should have a - fairly simple - algorithm to simplify parent directories", () => {
+    expect(resolveURL("http://toto.com/a", "b/c/d/", "torrent://g.a/b/c/d", "../a"))
+      .to.equal("torrent://g.a/b/c/a");
+    expect(
+      resolveURL("http://toto.com/a", "b/c/d/", "torrent://g.a/b/c/d", "../c/../../a")
+    ).to.equal("torrent://g.a/b/a");
+  });
+
+  /* tslint:disable max-line-length */
+  it("should have a - fairly simple - algorithm to simplify the current directory", () => {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Update or refactor this function so that its implementation doesn't duplicate the one on line 51. ([typescript:S4144](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4144))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=435&issues=AWgElM3n1_eUtAp8AFEC&open=AWgElM3n1_eUtAp8AFEC)

[ISSUE_KEY]: AWgElM3n1_eUtAp8AFEC